gui/static/js/components/scenarios.js

```diff
@@ -85,12 +85,35 @@ export default class ScenariosItem extends React.Component {
         var _this = this;
         var file = event.target.files[0];
         var reader = new FileReader();
-        reader.onload = function(event) {
-            var scenario = JSON.parse(event.target.result).items;
-            _this.setState({items: scenario});
-        };
+        var li = [];
+        var objectString;
+        _this.state.numBots = 0;
+        reader.onload = function(fileLoadedEvent) {
+            objectString = fileLoadedEvent.target.result;
+            var jsonArray = JSON.parse(objectString);
+            var regexPosition = new RegExp("\[\d,\d]");
+            jsonArray.forEach(function(object) {
+                //checks that bot contains position and angle
+                if (regexPosition.test(object.position) && object.type == "simulator.simbot" && object.position != "undefined" && object.angle != "undefined")  {
+                    _this.state.numBots++;
+                    console.log("bot:" + _this.state.numBots);
+                    li.push({type: object.type, angle: object.angle, position: object.position});
+                }
+                //checks that scenario object contains position, size, and angle
+                else if (regexPosition.test(object.position) && object.position != "undefined" && object.size != "undefined" && object.angle != "undefined") {
+                    li.push({type: object.type, angle: object.angle, size: object.size, position: object.position});
+                } else {
+                    alert("Invalid file! Please submit a properly formatted file!");
```

Comment:
why is there the alert inside the for loop? you should check to see if everything is valid, then if there is something that is not valid after you've gone over everything, then you alert once